### PR TITLE
Cc implement bundles

### PR DESCRIPTION
**Pull request by Chen2x**
Implemented the recursive step for displaying bundles for good mock server

### CODE REVIEW

**Comment by emre-f** [comment]
Is there a reason why we are `"0"` and `"1"` instead of `True` and `False` for the `is_bundle` property?

**Comment by Chen2x** [comment]
> Is there a reason why we are `"0"` and `"1"` instead of `True` and `False` for the `is_bundle` property?

I copied the DRS object over from starter kit DRS and followed its formatting. The spec doesnt say it needs to be 0 and 1 but I just kept it as is from starter kits. https://github.com/ga4gh/ga4gh-starter-kit-drs/blob/develop/database/sqlite/add-dev-dataset.sql

File: unittests/edit_data.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import json
 import os
+#import pdb;pdb.set_trace()
```
Comment (yash-puligundla) [inline unittests/edit_data.py:3]:
```suggestion
```

File: unittests/good_mock_server.py [hunk 2]
```diff
@@ -100,8 +101,15 @@ def get_service_info():
 @app.route('/ga4gh/drs/v1/objects/<obj_id>', methods=['GET','POST'])
 @conditional_auth(app.config["auth_type"])
 def get_object(obj_id):
+    expand = request.args.get('expand', default = False)
+
+    # convert param to bool
+    if expand == "True":
```
Comment (yash-puligundla) [inline unittests/good_mock_server.py:107]:
```suggestion
    if expand.lower() == "true": 
```

**Review by emre-f — DISMISSED** [review]
Everything looks good

**Comment by yash-puligundla** [comment]
@Chen2x Which DRS object should I use for testing if bundles work as expected. I tried "a1dd4ae2-8d26-43b0-a199-342b64c7dff6" but it gives a "No DrsObject found" error

File: README.md [hunk 3]
```diff
@@ -49,10 +49,23 @@ python compliance_suite/report_runner.py --server_base_url "http://localhost:500
...
+* credentials may be found for the associated auth type
+  * "basic" : config_basic.json
+  * "bearer" : config_bearer.json
+  * "passport" : config_passport.json
+
 ## Running the good mock server
 ```
 python unittests/good_mock_server.py --auth_type "none" --app_host "0.0.0.0" --app_port "8089"
 ```
+Make sure that the good mock server is running smoothely by making a GET request to
```
[the remark this answers is not in the document]
Comment (Chen2x) [reply README.md:63]: Yeah thats a typo I have updated on my end and will add to next commit

**Comment by Chen2x** [comment]
> 

That is the same object I have been testing with. I will try to see if I can recreate that error on my side and I will get back to you.

File: README.md [hunk 4]
```diff
@@ -49,10 +49,23 @@ python compliance_suite/report_runner.py --server_base_url "http://localhost:500
   * "basic"
   * "bearer"
   * "passport"
+
+Depending on the auth type selected, the appropriate credentials must be provided by the end user
+* credentials may be found for the associated auth type
+  * "basic" : config_basic.json
```
Comment (yash-puligundla) [inline README.md:55]: Add path to the files
```suggestion
  * "basic" : compliance_suite/config/config_basic.json
```
same for bearer and passport as well

**Review by yash-puligundla — COMMENTED** [review]
Hi Chen. I think we should rebase this branch on develop as there are commits on develop that are not in this branch.

**Comment by Chen2x** [comment]
> 

I think that is a good idea

**Comment by yash-puligundla** [comment]
@Chen2x 
1. I ran the good_mock_server and used postman to do GET "http://localhost:8089/ga4gh/drs/v1/objects/a1dd4ae2-8d26-43b0-a199-342b64c7dff6?expand=true". The child DRS objects are not expanded. In an ideal case, the child objects should be expanded
2. As per [DRS spec](https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.2.0/docs/#operation/PostObject), if auth = passports, then the optional expand param should be obtained from the request body and not the query param

File: unittests/good_mock_server.py [hunk 5]
```diff
@@ -129,6 +137,14 @@ def get_object(obj_id):
         if app.config["auth_type"]=="passport" and ("accept" in header_content and header_content["accept"] in [accept_type, "*/*"]):
             try:
                 request_body = request.get_json()
+
+                if "expand" in request_body:
+                    #if request_body["expand"].lower() == "true":
+                    #    expand = True
+                    expand = request_body["expand"]
```
Comment (yash-puligundla) [inline unittests/good_mock_server.py:144]: We should probably add a step to make sure expand variable is binary and not string. Also, I would remove the commented code in line 142, 143.

Comment (Chen2x) [reply unittests/good_mock_server.py:144]: Good call. Ill take a look at making these changes

**Comment by yash-puligundla** [comment]
Thank you @Chen2x Everything works as expected! 
Just one note, since there is a new field "is_bundle" in the drs object response, the report runner validation will always fail as we have set `"additionalProperties": false` in compliance_suite/schemas/v1.2/drs_object.json. 

We could either -
1. remove "is_bundle" property since it conveys the same informations as the presence or absence of contents array. (Referring to [DRS v1.2 openapi schema](https://github.com/ga4gh/data-repository-service-schemas/blob/master/openapi/components/schemas/DrsObject.yaml) -  If not set, this `DrsObject` is a single blob. If set, this `DrsObject` is a bundle containing the listed `ContentsObject` s)
2. Set `"additionalProperties": true` in compliance_suite/schemas/v1.2/drs_object.json. 

I lean towards option 2 since in practice, DRS implementers could also have added extra parameters and we don't want to penalize them for it.

**Comment by Chen2x** [comment]
> 

Hi yash I have made that change now. I agree with option 2 since Ian had mentioned a similar problem in the last cloud call of the fact that different drs implementers may have added additional fields.

**Review by yash-puligundla — APPROVED** [review]
Looks great, Chen! Approved and merging to develop.

**Pull request merged**